src/main/kotlin/org/opensearch/commons/alerting/model/BaseAlert.kt

```diff
@@ -0,0 +1,208 @@
+package org.opensearch.commons.alerting.model
+
+import org.opensearch.common.lucene.uid.Versions
+import org.opensearch.commons.alerting.util.IndexUtils.Companion.NO_SCHEMA_VERSION
+import org.opensearch.commons.alerting.util.instant
+import org.opensearch.commons.alerting.util.optionalTimeField
+import org.opensearch.commons.alerting.util.optionalUserField
+import org.opensearch.commons.authuser.User
+import org.opensearch.core.common.io.stream.StreamInput
+import org.opensearch.core.common.io.stream.StreamOutput
+import org.opensearch.core.common.io.stream.Writeable
+import org.opensearch.core.xcontent.ToXContent
+import org.opensearch.core.xcontent.XContentBuilder
+import org.opensearch.core.xcontent.XContentParser
+import org.opensearch.core.xcontent.XContentParserUtils
+import java.io.IOException
+import java.time.Instant
+
+/** CorrelationAlert and Alert can extend the UnifiedAlert class to inherit the common fields and behavior
+ * of UnifiedAlert class.
+ */
+open class BaseAlert(
+    open val id: String = Alert.NO_ID,
+    open val version: Long = Alert.NO_VERSION,
+    open val schemaVersion: Int = NO_SCHEMA_VERSION,
+    open val user: User?,
+    open val triggerName: String,
+
+    // State will be later moved to this Class  (after `monitorBasedAlerts` extend this Class)
+    open val state: Alert.State,
+    open val startTime: Instant,
+    open val endTime: Instant? = null,
+    open val acknowledgedTime: Instant? = null,
+    open val errorMessage: String? = null,
+    open val severity: String,
+    open val actionExecutionResults: List<ActionExecutionResult>
+) : Writeable, ToXContent {
+
+    init {
+        if (errorMessage != null) {
+            require((state == Alert.State.DELETED) || (state == Alert.State.ERROR) || (state == Alert.State.AUDIT)) {
```

Comment:
Why would alert state Deleted mandate a non-null error message

Reply:
yeah this one's in Alert state class also, so eventually Monitor based Alert (i.e Alert class) will also extend this BaseClass, therefore.